src/Language/PureScript/Ide/State.hs

```diff
@@ -246,6 +232,71 @@ resolveLocationsForModule (defs, types) decls =
         annotateType x = IdeDeclarationAnn (ann {_annLocation = Map.lookup (IdeNamespaced IdeNSType x) defs})
         annotateKind x = IdeDeclarationAnn (ann {_annLocation = Map.lookup (IdeNamespaced IdeNSKind x) defs})
 
+convertDeclaration'
+  :: (P.Ident -> IdeDeclaration -> IdeDeclarationAnn)
+  -> (Text -> IdeDeclaration -> IdeDeclarationAnn)
+  -> (Text -> IdeDeclaration -> IdeDeclarationAnn)
+  -> (Text -> IdeDeclaration -> IdeDeclarationAnn)
+  -> IdeDeclaration
+  -> IdeDeclarationAnn
+convertDeclaration' annotateFunction annotateValue annotateType annotateKind d =
+  case d of
+    IdeDeclValue v ->
+      annotateFunction (v ^. ideValueIdent) d
+    IdeDeclType t ->
+      annotateType (t ^. ideTypeName . properNameT) d
+    IdeDeclTypeSynonym s ->
+      annotateType (s ^. ideSynonymName . properNameT) d
+    IdeDeclDataConstructor dtor ->
+      annotateValue (dtor ^. ideDtorName . properNameT) d
+    IdeDeclTypeClass tc ->
+      annotateType (tc ^. ideTCName . properNameT) d
+    IdeDeclValueOperator operator ->
+      annotateValue (operator ^. ideValueOpName . opNameT) d
+    IdeDeclTypeOperator operator ->
+      annotateType (operator ^. ideTypeOpName . opNameT) d
+    IdeDeclKind i ->
+      annotateKind (i ^. properNameT) d
+
+resolveDocumentation
+  :: ModuleMap P.Module
+  -> ModuleMap [IdeDeclarationAnn]
+  -> ModuleMap [IdeDeclarationAnn]
+resolveDocumentation modules =
+  Map.mapWithKey (\mn decls ->
+    maybe decls (flip resolveDocumentationForModule decls) (Map.lookup mn modules))
+
+resolveDocumentationForModule
+  :: P.Module
+    -> [IdeDeclarationAnn]
+    -> [IdeDeclarationAnn]
+resolveDocumentationForModule (P.Module _ _ _ sdecls _) decls = map convertDecl decls
+  where 
+  comments :: Map P.Name [P.Comment]
+  comments = Map.fromListWith (flip (<>)) $ mapMaybe (\d -> 
+    case name d of 
+      Just name' -> Just (name', snd $ P.declSourceAnn d)
+      _ -> Nothing)
+    sdecls 
+
+  name :: P.Declaration -> Maybe P.Name
+  name (P.TypeDeclaration d) = Just $ P.IdentName $ P.tydeclIdent d
+  name decl = P.declName decl
+
+  convertDecl :: IdeDeclarationAnn -> IdeDeclarationAnn
+  convertDecl (IdeDeclarationAnn ann d) = 
+    convertDeclaration'
+      (annotateValue . P.IdentName)
+      (annotateValue . P.IdentName . P.Ident) 
+      (annotateValue . P.TyName . P.ProperName)
+      (annotateValue . P.KiName . P.ProperName)
+      d
+    where
+      docs :: P.Name -> Maybe Text
+      docs ident = convertComments =<< Map.lookup (ident) comments
```

Comment:
I'd prefer a `fromMaybe ""` at the end of this, so it's easier to tell whether or not we've already attempted to resolve documentation.

Reply:
So returning `""` instead of `null` in the json output if there is no docs? Maybe `Maybe Maybe _`?

Reply:
I thought about `Maybe Maybe` but at that point I'd prefer a custom three valued data type. Also note that `convertComments` returns `Nothing` if it sees an empty string.

Reply:
Ahh, I see it now the `KiName` `TyName` ... stuff is from Docs. Sorry for being a little thick xD

While we're at it, there are redundant parens around `ident` in this line.